### ISSUE TITLE
Images to Act On - How Should It Work?

### ISSUE THREAD

**Issue by elstoc** [darktable-org/darktable#6025]
Now that a lot of the decision making around image selection in the lighttable and thumbtable is centralised into a single function (``dt_view_get_images_to_act_on``) I thought it might be time to talk about how it works and come to a consensus on how we can make it a little more user-friendly.

Here's now it works right now:

```
                mouse over| x | x | x |   |   |
        mouse inside table| x | x |   |   |   |
    mouse inside selection| x |   |   |   |   |
             active images| ? | ? | x |   | x |
                          |   |   |   |   |   |
                          | S | O | O | S | A |
     S = selection ; O = mouseover ; A = active images
     the mouse can be outside thumbtable in case of filmstrip + mouse in center widget
   
     if only_visible is FALSE, then it will add also not visible images because of grouping
```

I find myself constantly consulting the above table and still have trouble keeping it in my head. And I don't think I'm the only one who sometimes finds it confusing. Given the number of bug reports and discussions where this comes up it may be better if it were a little more simple/intuitive or perhaps if some of its behaviour could be simplified via preference settings or flagged better in the UI.

For example, probably the most commonly used mode in the lighttable is called 'file manager'. Most users would probably expect it to work (within reason) like a standard file manager and in this mode at least, being able to perform actions just by hovering over images and hitting key combinations is decidedly non-intuitive. And it gets even odder when you have multiple images selected but are inadvertently hovering over an unselected one.

Anyway, I've just opened this for discussion to see whether there might be a better way.

**Comment by johnny-bit** [comment]
Well, It's mostly intuitive except for mouseover - so if mouse is inside selection, the get images to act on will return whole selection. if mouse is hovering over different image then (uh... correct me if i'm wrong) it'll return only the mouse-hovered one?

So the rule IMO should be: if there's a selection going on preffer selection over mouseover and it'd prooobably save much headeaches.

**Comment by aurelienpierre** [comment]
I have always found the behaviour of the lighttable confusing. If you select a thumbnail (click), then hover the mouse somewhere else, then move arrow keys, it's the hovered thumbnail that gets updated.

I think selection (for edit or export) should rely on click only, and hover should only be used to display transient info (read-only) like metadata. Mouseover should be ditched for everything else. Selection (actively and explicitely done) should be the only reference for every export, edit or action triggered in GUI.

**Comment by elstoc** [comment]
> I think selection (for edit or export) should rely on click/arraw only, and hover should only be used to display transient info (read-only) like metadata. Mouseover should be ditched for everything else. Selection (actively and explicitely done) should be the only reference for every export, edit or action triggered in GUI.

Totally agree. My understanding though is that arrow key (to navigate) doesn't change selection at the moment but acts the same as a 'mouse hover'

**Comment by junkyardsparkle** [comment]
Some people seem to really like the hover-selection, so this will probably be contentious. Personally, I've been hoping for an option to disable it from day one. I would also think disabled would be a safer default for new users, but that's just me. For heavily mouse-oriented people it isn't as much of a problem, because placing the pointer on a button to execute an action inherently moves it off of the thumbnails. For keyboard oriented people, it's insanely dangerous.

**Comment by AlicVB** [comment]
> Now that a lot of the decision making around image selection in the lighttable and thumbtable is centralized into a single function (`dt_view_get_images_to_act_on`)

Yep, this as been done with that in mind (not only)

Now, for the rest of the discussion, I agree that this behavior may be confusing for some people and not just newcommers which may be confused by far more than that anyway ;)
On the other hand, the current behavior can be really handy once you are used to it. I use that quite often and I know for sure that other users and devs use i it a lot too.

What I would propose is to have a pref to switch to another "image to act on" algorithm.
For defining how it should work, I think the table mentioned in first post can be a good starting point. The delicate area is -of course- the filmstrip...

**Comment by elstoc** [comment]
There are also probably some view-specific special cases we need to consider. For example

- If a single image is selected but it's not currently visible, perhaps operations should not consider it as being selected (user might just have clicked without meaning to select)
- If lighttable view shows only one image at a time, selections should be ignored and actions should be carried out on the currently viewed image whether or not the mouse is hovered over it

**Comment by elstoc** [comment]
I'd be interested to know what 'Active' means in the table above.

**Comment by jobec** [comment]
Like mentioned above, hovering can be powerful and allows for very fast actions. But it should not be the default.
E.g. if anything is selected (by a simple click, ctrl+click, shift+click etc etc), hovering should be totally disabled and do nothing.
If nothing is explicitly selected (by single clicking on a selected image to unselect it), mouse hovering is what defines the selection.

I'm not sure if my explanation is clear, but I think that'll be most intuitive and match with how file browsers or even digikam works.

**Comment by AlicVB** [comment]
> I'd be interested to know what 'Active' means in the table above.

active images are the one(s) shown in central view in modes with filmstrip (preview, culling, darkroom)

**Comment by RawConvert** [comment]
This is still happening in 3.4.
There have been a number of issues raised about it which seem to have been distilled into this one.
I think it deserves a decent push towards a fix!

**Comment by jobec** [comment]
I left darktable behind in the meanwhile.
I got so fed up with all the quirks in the GUI that I spent more time clicking around then actually working on images.
I know some think the way it works is a "feature" and that you get used to it, but it is not. It deviates from the unwritten and implicit GUI rules all software seems to use.

On a sidenote: one other thing that pushed me away was the filmic RGB stuff. It's a brilliant module, don't get me wrong, but I never managed to get the results I wanted and never really understood the logic behind it. And I don't want to become an image scientist just to know how to use something.

I find this a pity, because the potential of darktable is gigantic and I am a heavy fan of opensource. Maybe I revert back to it some day...

**Comment by RawConvert** [comment]
@jobec , each to their own of course.  I think dt is very good overall, but could be a step better if various basic quirks and bugs were sorted out.

**Comment by elstoc** [comment]
@RawConvert we need to come to some consensus as to the way forward before we can progress to coding it. Right now it's all working "as designed" but we need to decide if we want to change the design or maybe implement some different alternatives that are selectable as a user preference.

**Comment by Nilvus** [comment]
> @RawConvert we need to come to some consensus as to the way forward before we can progress to coding it. Right now it's all working "as designed" but we need to decide if we want to change the design or maybe implement some different alternatives that are selectable as a user preference.

I think having an alternative could be good. And the behavior should be by default consistent with other applications to help new users use darktable. So, something with proposal in #8898 and @aurelien [comment](https://github.com/darktable-org/darktable/issues/6025#issuecomment-676609742) above should be default one, and actual one an alternative. Probably the simple way. If this could be implemented in 3.6, that would be great and limit new issues about that. But of course, it remains few time to do that.

**Comment by RawConvert** [comment]
hi,  my eyes have had a problem and I've been out of things somewhat, anyway getting better now and in fact built a new dt a few days ago.  I don't know why I never responded to @elstoc in Jan, apologies for that.  So here is my 2p worth...
I'll re-iterate that I think things like hover-to-delete, and ignoring a selection in favour of a hover, are wrong.  I agree with comments above along the lines of behave like a normal file manager and just use hovering for pop-up info.
I don't use the filmstrip and always use lighttable in "file manager" mode, so am not qualified to comment on functionality like culling.
@elstoc said above "_If a single image is selected but it's not currently visible, perhaps operations should not consider it as being selected (user might just have clicked without meaning to select)_"   Going back to normal OS file manager, it is surely fine to select a subset of files and scroll down to select more.  They may go off the screen but you explicitly selected them so they are acted on.  Same with lighttable surely?
Also "_If lighttable view shows only one image at a time, selections should be ignored and actions should be carried out on the currently viewed image whether or not the mouse is hovered over it_"   Sometimes I only have one image loaded, I'm in darktable, back to lighttable to export, but it's not selected, surely in this situation it should be?
Having a preference to control how mouseover works seems ok providing the default is "mouse does popup info only".  However settings like this presumably contribute to code complexity and scope for bugs?

**Comment by AlicVB** [comment]
As said earlier, the code has been done to allow more easily multiple implementations of dt_view_get_images_to_act_on
Personally, I'm very pleased with the actual implementation, but I can understand that other would prefer a more "classic" approach...

Lighttable filemanager in zoom > 1 is the easy part.
What is challenging is the other modes, the interaction with filmstrip, the acceptance (or not) to loose or complexify some features (like multiple copy paste)
An example with culling with 2 images : you have to choose what happen when : 
- the mouse is outside the central area and the filmstrip
- the mouse hover an image in the central area with 2 cases : selected or not
- the mouse hover an image in the filmstrip, with 2 cases : selected or not

We also have some actions that can only take one image into account : copy, 'd' shortcut, etc...
All this needs to be consistent between views / cases...

Of course, I'm ready to help listing all this if someone is brave enough to step in ;)

**Comment by AlicVB** [comment]
> Is that hard to do?

Sadly, yes... because there's features that rely on it : metadata view is an example, but I'm sure we can find lot of others. There's also cases where we just can't ignore hover : culling mode is an example : if you hover an image and press 'r' : you don't expect to reject all your selection...
From the code point of vue, that would mean writing an alternative `dt_view_get_images_to act_on` which is not a problem but need to be though carefully...

**Comment by aurelienpierre** [comment]
I maintain what I said a year ago. Very often, because I kick the mouse or because palm rejection on touchpad still sucks on Linux, I end up with an unexpected file being exported or rated (with the stars). "Act on hover" is simply brittle. Acting should be done after clicking on something hard (button or key), it's simply too easy to have an erratically moving cursor.

**Comment by AlicVB** [comment]
> But what does it mean if there is a [X],[space],[?] in column 'O' and why do we have O and S twice?

It's my work, so apologize if I were cryptic...
x : condition validated
space : condition not validated
? : condition validated or not

For the bottom line, it's the image(s) we return... effectively, there is cases where we return the same set of images...

read the table column by column from left to right : 
- if col1 is validated then return selection (S)
- otherwise if col2 is validated return hovered image (O)
- otherwise if col3 ...

**Comment by Solarer** [comment]
Thanks, now it makes sense.
But isn't
```
selection with mouse over > mouse over > active > selection
```
A simpler but equivalent representation?
Where A > B means that B will not be used if A is available.

**Comment by Solarer** [comment]
Here is a list of related issues that have been posted in the past:

* #6064: General confusion over mouse over + selected behaviour.
* #6313: Behaviour of 'mouse over + selected' pictures is unintuitive.
* #6625: When entering culling mode via 'X' button, accidental movements of the mouse or even the starting position might change whether actions are applied to all active pictures or only one (https://github.com/darktable-org/darktable/issues/6625#issuecomment-740207408).
* #7254: Images in culling mode cannot be selected with a single left click as in filemanager & Mouse over is not always plain visible and therefore prone to errors.
#7805: People find it unintuitive that you cannot select something in culling mode with a regular mouse click (related to #7000 and #5138)
* #8898: Selection has no effect in culling modes because 'active' images always have higher priority.

So it seems to come down to the following points:
a) The current interaction between mouse over and selection feels inconsistent and unintuitive
b) The 'priority chain' of ``dt_view_get_images_to act_on``` is problematic in culling (fixed) mode because at least one picture is always active and has higher priority than any selection.
c) Working with selections in culling modes is completely different than in any other mode
d) The experience is not always clean when using mous/keyboard/mouse+keyboard

**Goals**
The solution needs to be consistent throughout all lighttable and darkroom modes and should ideally be intuitive, so that it can be easily adapted by new users. It must support all common (and reasonable!) workflow steps with mouse, keyboard and mouse+keyboard. Safety against unintentional actions might be an additional goal.

I most likely forgot something so feel free to comment.

**Comment by RawConvert** [comment]
@Solarer , good to see someone taking a bit more of an overview on this, to complement the code-oriented approach.
Where you say under Goals about being consistent and intuitive, I'd say reasonable compatibility with your average file manager is a good goal.  I don't think we should emphasise new users too much, I think lots of users find the behaviour odd who are not new users, e.g. Aurelien P!  And I think safety is not just an additional goal, it's pretty important.
I think there are other issues you could add to your list - #4558, #5895, #5564 possibly, #5987, #5984...

**Comment by Solarer** [comment]
@RawConvert thanks for that. I added all of your issues except #5564 because it seems to be unrelated and might have been fixed already since the issue is from 3.0.1 and I was unable to reproduce on master.

**Comment by Solarer** [comment]
**Draft for a) : Mouse-Over**
Overall, I think that mouse over is a valuable shortcut because it saves you an additional mouse click per action. That being said, I don't think that mouse over should ever take priority over selection. 

I drafted a few cases in which selection + mouse-over could interact ('change rating' is just an example):
* You selected some pictures to change rating and hover over an image by accident
* You selected some pictures to change rating and want to include another picture in that same action
* You selected some pictures to change rating and realise that you want to do some other (unrelated) action on another (unrelated) picture first. For example: set a color label on that one picture.

Scenario 1 is probably the most likely and the reason why we have so many open issues. In scenario 2, the additional picture could just be added to the selection. Scenario 3 is the only one in which the current implementation provides an advantage. But how often does that happen and I don't know a single filemanager that behaves that way. This makes this feature very unintuitive. To make things worse, if you hover over an image that has already been selected, the whole selection is chosen instead of the picture you hovered over. So in scenario 3 I would expect the user to apply the rating first and then deselect all pictures before dealing with the color rating. He then can use mouse-over or selection to change the color label.

PS: Some people might still want to disable mouse-over completely because they don't like it or deem it dangerous. I am not sure if that is still necessary if selection takes priority over mouse-over but that could be discussed as well.

**Comment by Solarer** [comment]
**Draft for b) and c) : a change to culling**

Now, this is not only about changing ```dt_view_get_images_to act_on``` but I also propose a change for culling in general.

Culling officially has two modes (fixed and dynamic) but to be honest with you, in reality there are 3 modes:
1. If you enter culling (fixed) with selecting 0 or 1 images, you can scroll though all pictures without limitation and if you select more picture after entering the mode, that does not change. So in this case culling (fixed) behaves very similar to the filemanager. (I use this mode all the time because I prefer to rate and reject pictures like this rather than the filemanager mode. This is also how Lightroom does it.)
2. If you enter culling (fixed) while having selected 2 or more pictures, you can only scroll through your selected pictures. If you select more pictures, you can scroll through these as well and if you deselect them, you cannot scroll to them anymore. This is very different from the first mode because you cannot use selections as you are used to since releasing your selection locks you in place.
3. Today's dynamic mode. This is the same as 2 with the only exception that you always see **all** selected pictures. You cannot enter this mode without selecting at least one picture.

My proposal is to make a real split between modes 1|2 and instead merge 2&3 because they already are more or less the same. The only difference between 2&3 is how many pictures are displayed. So today's dynamic mode (3) could simply become position '0' in the slider of mode 2 where you select how many pictures you want to see. So it could look like this: [dynamic, 1, 2, 3, ...].

The first culling mode could then behave exactly like todays filemanager including the ability to select a picture with a single mouse click and adding more pictures with a Ctrl-click. Following the same arguments as in my previous post, selection should take priority over mouse-over and mouse-over takes priority over active images (We did not have active images in filemanager). That would tackle #6625, #7254, #7805 and #8898

In the second culling mode, all relevant pictures are always selected. So the priority has to be hover > active >selected. (With active=selected for the dynamic setting). 

Please let me know what you think. This is probably the most controversial topic and while typing this I realised that I have no idea what other people use the culling modes for.
I always use the first culling mode for my initial selection, rarely use the second to compare 2 pictures and never use the third (dynamic) because I can't deal with having more than 4 pictures on the screen at once.
**If people use the culling modes for other things, that would break because of this proposal, please post that here.**

**Comment by Solarer** [comment]
**Draft for the rest**
* In full preview, only the displayed image should be acted on - independent of what might have been selected in the background and independent of the mouse position.
* In darkroom, the priority should probably also be ```selected > mouse-over > active``` just like my suggestion for filemanager and the new culling. I am not 100% sure, though.
* Mouse and keyboard support in all modes: needs to be tested.

**Comment by AlicVB** [comment]
Lot of work, @Solarer. Thanks !
Some thoughts : 

First : we need to ensure to avoid breaking any workflows. While offering multiple image_to_act_on algorithm can allow that easily, I fear that your proposal about culling may be way more difficult to handle or ends with a "spread everywhere" change, which will be very difficult to maintain.

Now about your ideas, some questions that are not handle here (or maybe I miss them) : 
- which image if pressing "single" image shortcut (example 'd')
- how about multiple copy-paste (again with shortcuts) ?
- how about selection when entering full preview or darkroom ?
- is there a difference if the mouse is on filmstrip or in the main window, or outside ?

I'm not convinced about your culling proposal. Culling goal is to compare a set of similar images. It has some advanced features to do that (automatic positioning, zoom/pan, ...) While it can be used for other things too (like your workflow) we should keep the initial goal in mind !
for me, 1 and 2 are the same, except that for 2, we limit the set of images (possibly with non continuous images selection) which is handy ! (this also works for full preview)
3 is completely different (as you state). The main usecase is when you want to find quickly the best images of a set of similar images : you show them all at first, then reject the most obviously bad and so on until you have only one image.

I'm sorry if that sound too critical, it's just that I know pretty well this part of the code, and so exactly know what can break and where ;)

**Comment by Solarer** [comment]
Your input is much appreciated. Everybody is more versatile in a different area, so concerns should be raised.
I totally forgot about one image shortcuts :(

Can you elaborate on "multiple copy-paste with shortcut"?

**Comment by Solarer** [comment]
>  "copy history from 1 image" followed immediately by "paste this to multiple images (the selection)"

That would still work but you would need to hover over an image to copy its history **first** and **then** select your target images to paste. The other way around is not really natural. Think about a file manager. You would not lock in the paste destination first and then go back to select the pictures to copy. It works the other way around.
Or do people use this to select some target pictures and then take some edits from here and some edits from there, basically combining edits from multiple pictures?

> I fear that your proposal about culling may be way more difficult to handle or ends with a "spread everywhere" change, which will be very difficult to maintain.

That I cannot evaluate, since I am not too familiar with the code. But from my point of view, 3 is just a special case of 2 where number_of_active_images = number_of_selected_images. Automatic positioning, zoom/pan, ... behave completely identical in both modes. You can try it for yourself: Select ~12 images, enter culling (fixed) and increase the slider to a high number such as 16. Now you can switch back and forth between fixed and dynamic mode as you deselect picture after picture. Both modes behave identically 'dynamic'. Based on that, both pieces of code should be identical except for a single ```if``` clause that is needed for setting number_of_active_images = number_of_selected_images in mode 3.
In mode 1 however, you can scroll past your selection, which is impossible in 2 and 3.

> which image if pressing "single" image shortcut (example 'd')

I totally forgot that and you are right, an exception should be made for at least some of the shortcuts (maybe all). For example it should still be possible to switch to full screen with 'w' when I hover an image while a selection is active in case I need to check a picture. About 'd' I am not sure - but it could be done since it does not really 'break' anything if you hover an image by accident.
Then again, if you select 1 picture and move your mouse by accident before hitting 'd', would you not expect your selected picture to be shown in the darkroom? :P 

> how about selection when entering full preview or darkroom ?

If you don't hover anything, first selected picture should be taken (maybe first active in culling mode). 
If you hover something, that's open for discussion. Could be either first selected picture or hovered picture. Either way the selection should be maintained.

> is there a difference if the mouse is on filmstrip or in the main window, or outside ?

I would only differentiate between mouse-over or no mouse-over no matter where the picture is.

**Comment by AlicVB** [comment]
> Think about a file manager.



> would you not expect your selected picture to be shown in the darkroom? :P

No I wouldn't... because I don't refer at how filemanagers works when I use dt. That's certainly because I'm an very old user ;)



> Or do people use this to select some target pictures and then take some edits from here and some edits from there, basically combining edits from multiple pictures?

yes, all the time, esp in other modes than filemanager : you do your selection in filemanager (clearly the only mode with zoomable when you can do that for real) then you go, for example in darkroom, do some editing and copy/paste it to the selection.
Same apply for culling (but less frequently)



> That I cannot evaluate, since I am not too familiar with the code. But from my point of view, 3 is just a special case of 2 where number_of_active_images = number_of_selected_images. Automatic positioning, zoom/pan, ... behave completely identical in both modes.

I know, that's my code (originally) :) in fact all culling and preview modes share the same code. What has been chosen (for all modes, not only culling) is to not differentiate based on the input selection, but on how it works. The selection (if present) is only here to limit the set of images (you can imagine that as a sub-collection) That has been done to get similar workflow when you want to traverse you whole selection and when you are only interested in a subset of it...



> I would only differentiate between mouse-over or no mouse-over no matter where the picture is.

If I understand, that may have strange results : imagine you have a selection of 50 images, you are in culling with 2 images shown and you use whatever left panel lib to apply something... imho, user expect that the action will be done on the 2 main images... (not to speak if filmstrip is hidden)

> Either way the selection should be maintained.

agreed, imho the selection should always stay, expect on intentional user action (ctrl-click, ctrl-a, ...)

**Comment by Solarer** [comment]
Thanks again for your feedback!

> because I don't refer at how filemanagers works when I use dt

I was making that reference because of @RawConvert's comment from a few days ago which I find very reasonable:
> about being consistent and intuitive, I'd say reasonable compatibility with your average file manager is a good goal.

#------------
> yes, all the time

But could it not be done in the opposite order? Let's say I have a bunch of pictures that are all similar. I would edit one of them, copy that image's history stack, go to filemanager, select all similar images and then paste.
Of course, changes to ```dt_view_get_images_to act_on``` will require some people to adapt how they interact with pictures - especially those that have been around for a while and know how to utilize dt's little quirks. But if the new implementation only changes the order in which things are done but does not require a ton of extra clicks or button-presses while being more consistent and intuitive, then that is probably a step in the right direction.

> you can imagine that as a sub-collection

I understand that and while I do not use it a lot, I see how valuable that is and I would not want to take that feature away. I would simply reorganize how you enter culling modes 1, 2 and 3 and help visualize in what mode you are currently in! 
At the moment, you can only choose between modes 2 and 3 via gui and you end up in mode 1 by accident if you did not have a selection when you wanted to enter mode 2. In the same manner you end up in mode 2 by accident if you had a selection while you wanted to enter mode 1.
Visually, you can not differentiate whether you are in mode 1 or 2 - you need to scroll and check if you can scroll past your selection to tell the modes apart. Additionally, if you ended up in the wrong mode, there is no way to switch back. You need to leave culling, change your selection and then reenter culling. 
Hence I would split 1 and 2 to keep them visually apart and stop the accidental switches. And then optionally merge 3 as a special case of 2 so that we do not need more than five gui buttons for the modes. Functionality would stay the same.

> imho, user expect that the action will be done on the 2 main images

I agree for modes 2 and 3 since you are working on a 'sub-collection' as you call it. Because of that you cannot use your selection anymore and that more or less turns your active images into your only way of 'selecting' pictures here. So in these modes, your actions should be applied to all active images.
In mode 1 you are not limited to a 'sub-collection' and can select images as you like. Therefore I would prioritize selection over active here. That would probably solve the complaints about selection not working in culling for export/ratings/etc. 

Best, Solarer

**Comment by johnny-bit** [comment]
There are voices from users about unintentional hovers, eg: https://www.reddit.com/r/DarkTable/comments/nijve1/preventing_tag_editing_on_hover/

**Comment by Nilvus** [comment]
Should be good to find a good way with that on so close this debate for 3.8.

**Comment by RawConvert** [comment]
@Nilvus , just curious why 3.8?  Master is 3.5 at the moment I think, so 3.8 seems an awful long way away.

**Comment by elstoc** [comment]
@RawConvert: 3.6 is due to be released at the start of July. We're way past feature freeze now.

**Comment by Nilvus** [comment]
> @Nilvus , just curious why 3.8? Master is 3.5 at the moment I think, so 3.8 seems an awful long way away.

As @elstoc says. It's too late for 3.6.

**Comment by Solarer** [comment]
@RawConvert  3.7 will be skipped, so 3.8 is the next release. Will be a nice Christmas present :)

**Comment by RawConvert** [comment]
Ok!  Best wishes devs for a good fix to this issue.

**Comment by alxgoldstn** [comment]
New user here... I find hover selection to be extremely unintuitive. Perhaps some software works like this, but it's not how any software I've ever used works. I actually thought it was a bug and was getting super frustrated with the program.

Side note: Honestly the UI in DT is probably it's biggest issue for conversion (because the rest of it is great.) I've been struggling with picking up DT and keep googling for other software because I keep finding it's way too difficult to use the software intuitively. Read reviews of it online and this is also the number 1 issue people mention.

**Comment by Solarer** [comment]
I also can't wait until we get started here but I think people are still busy with the release - so let's give them some time. 
In the meantime you can leave feedback to my draft if you want. I don't think that we have agreed on an approach yet and that needs to be done before the coding starts.

**Comment by Solarer** [comment]
@here Now that 3.6 is out and most urgent bugs seem to be fixed, should we have a second look at this and start working?

**Comment by TurboGit** [comment]
Moving to 4.0 but clearly would be nice to have a summary about what remains on this issue as it is long and convoluted.

**Comment by Solarer** [comment]
I don't think anything has been done yet so what remains is a draft coding and see how it feels like and then iterate until all cases are covered

**Comment by ianhattendorf** [comment]
I'd like to throw #10592 in for consideration

> Expected behavior
Able to copy history of RAW file while image grouping is enabled. Similar to how double-clicking a grouped image opens the RAW file or mousing over a grouped image shows the RAW info, I would expect history copy to copy the history of that RAW file.

**Comment by Solarer** [comment]
Hi @AlicVB,
I was hoping to code something during the holidays and as in preparation for this I was looking into `src/dtgtk/culling.c`
Can you explain to me some phrases in the comment block lines 888-902?

What is "first image" in line 891
What is "first" in line 892 ("first selected" is clear, but what is "first"?)

Maybe it will become clear to me once you explained it, but what do I need to do in darktable to reach the states described in lines 896 and 901?

Thank you!

**Comment by AlicVB** [comment]
I wonder who has done such incomprehensible comments :)
So, let's try to remember...
I would propose to change the comment to something like : 
```
/** HOW it works :
   *
   * first image to show :
   *  offset OR image_over OR first selected OR first image in thumbtable view OR -1
   *
   * "navigation only inside selection" state :
   *  culling dynamic mode                       => OFF
   *  first image is in selection AND selection length > 1 => ON
   *  otherwise                                  => OFF
   *
   * "selection following" state :
   *  culling dynamic mode                                    => OFF
   *  first image(s) == selection && selection is continuous  => ON
   */
```
is that better understandable ? I you still need some explanation, that mean the comments need further changes ;) otherwise I will propose a PR with that change.

**Comment by Solarer** [comment]
Hope everybody had peaceful Christmas days!

I found some time to implement a new version of images_to_act on but there is still one problem that I am stuck with and I would like to sort it out before I share the code.
I have to check if I am in culling-dynamic mode because the code needs to behave differently there and I do it like this:
`dt_view_lighttable_get_layout(darktable.view_manager) == DT_LIGHTTABLE_LAYOUT_CULLING_DYNAMIC`

What I realised is that this function also returns true when I am in the darkroom if I entered it from culling-dynamic.
To me this makes no sense, it should return false in darkroom, map or other layouts. Is this a bug? If it is not a bug, why is it implemented that way and what is the best way for me to check if I am in darkroom or lighttable from within images_to_act_on?

**Comment by AlicVB** [comment]
well, I don't see this as a bug : as the name of the function indicate, it's the layout in lighttable which is returned...
if you want to know the current view, you can use something like : 
```
const dt_view_t *v = dt_view_manager_get_current_view(darktable.view_manager);
if(v->view(v) == DT_VIEW_DARKROOM)
{
  ...
}
```

Another point to avoid you too many refactoring, have you seen #10649 ? I've refactored that part of code in a separate file, so implementing a parallel algorithm should be simpler and more clear...
(I've also fixed and enhanced the caching, but that shouldn't change anything for you)

**Comment by Solarer** [comment]
Thanks, it worked fine.

No I have not seen #10649. I created a branch ~12days back so we will need to do some merging at some point but first we need to align if people are happy with the new approach. I will share the code tomorrow ;)
Should I do a pull request or link to my repo?

**Comment by AlicVB** [comment]
pull request is better imho, you'll have better test field and it's easier to review/comment

Note that, as I've said you'll need to do some rebasing first as PR #10649 has just been merge into master. Basically, you have to move everything to `act_on.c` just ask if you need help. Sorry for the extra work... but that should be cleaner to read now !

**Comment by Solarer** [comment]
Thanks, I think I managed the merge without messing things up :P
Here is the PR:  #10728

**Comment by ChristianBirzer** [comment]
As a new user (and hopefully soon developer 😉) to dt I'd like to add a different approach how to handle selections and hovers to the discussion (and sorry for again comparing a little bit with LR... I just can't stop 🤣):

What about separating actions to selected images and the hovered image to different modes? I think about something like LR does with its spray can. Some sort of point-and-shoot mode?

By default, the selection-mode is activated in file manager and filmstrip. So all actions go only to the selected images. If no image is selected, nothing is changed. Hovering is ignored. Like I would expect as a new user.
(I'd exclude the 'w' shortcut for zooming into an image because this does not change anything and is very handy)

By activating the point-and-shoot mode (find a better name!) the selection is cleared (no selection possible at all) and all shortcut actions affect only the hovered image. This would allow to very fast reject or rate images. I can also imagine a couple of shortcuts to add some predefined tags to the hovered image ("favorite tags").
In culling mode, it already seems that there is no selection possible (at least in the central widget), so this would always use the point-and-shoot mode.

By separating the problem into two different modes, I think it would be friendlier for new users (they simply work with the selection mode until they discover and explicitely activate the other mode) but still allows to use the quick mode for experienced users without getting confused about selection/hover/hover-in-selection stuff..

I'm sure, I've overlooked a lot of things that argue against it, but from a user's point of view it would simplify some things.

**Comment by Solarer** [comment]
That is more or less how my code behaves behind the scenes. If some pictures have been selected, hover is disabled (you call this selection-mode) if no pictures are selected, hover works. W works always because it does not change anything.
So there is no button to "switch the mode" but you do it by using a selection or not.

**Comment by phweyland** [comment]
Looking at #11589, the issue can be summarized as:
if selected images and occurs hovering, no validation (apply) should happen (because we would apply on the wrong image).
Then it would interesting to get from  dt_act_on_get_images() the information:
- hovering  and no selection
- hovering and selection
- selection only

This way the module would be able, at least, not to make a wrong action.

**Comment by Solarer** [comment]
@phweyland 
Or (since you plan to ignore the hovering anyways), we just return the selected images ;) 
Problem is: current consent with @AlicVB was that we do a dual implementation because some people prefer the current images_to_act_on. That means we need another special case handling in the metadata module - creating even more spaghetti code.
I am still in favor to fix this implementation for all users in the same way. It is too cumbersome to maintain the old implementation. There will be even more cases like this

**Comment by phweyland** [comment]
@Solarer, the hovering is still very useful to show information.

**Comment by Solarer** [comment]
I think I did an overwrite so that showing information works as before using hover.

**Comment by phweyland** [comment]
In any case I would vote (I think I've already) to use hovering only to show information and not to apply anything. I wouldn't mind to have a preference if we wanted to keep the feature (more spaghetti, though :)).

**Comment by johnny-bit** [comment]
Hover to show info (like say tooltip or in info modules) and only operate on selected items (and hovering NOT implying selection) is imho best way forward. And less spahetti please :)

**Comment by Nilvus** [comment]
I titally agree with you @johnny-bit. And good to see you again. Was a long time.

**Comment by mlaverdiere** [comment]
Late to the game, but to be short, while I love dt, I'm also experiencing problems with lightable hovering/selection process. Each time I do some culling with dt, I end up with a lot of errors when applying tags, stars, colors, rejection, etc.

I suppose that what @johnny-bit and @phweyland are proposing makes sense (hovering for info only), but I would even prefer the behaviour of a regular file manage (like GNOME Files/Nautilus), where hovering does nothing (except some subtle icon highlighting).

**Comment by RawConvert** [comment]
I also agree with johnny-bit on 22May.

**Comment by quovadit** [comment]
I agree that it's the best solution not to operate on hovered items by default.

But there are situations, where it's very convenient to quickly apply color-ratings on a number of images without having to click everytime.

That's why I suggested two different modes:
- per default:
  - all actions are applied on the selected image(s)
  - if no image is selected, a toast-message is displayed: no image(s) selected
  - pressing an arrow-key changes the selected image (like nautilus)
- 'point-and-shoot'-mode:
  - a new toggle-button in the top bar (reticle) is activated
  - all actions are applied to mouseover
  - so all buttons in 'selected images' are disabled


See details and mockups in https://github.com/darktable-org/darktable/pull/10728#issuecomment-1021558133

**Comment by rekcodocker** [comment]
Ah, so that is where this behaviour comes from.  It's not a bug, it's a feature.  Or actually (and that's the problem) many features in one. 

I notice in the discussion that some people want to have as many ways of selecting/acting as possible.
However, even though you can explain it, it is not intuitive. 
The other approach is to work from what a user might expect - and trust me, on a filemanager-layout she expects filemanager behaviour.

This would work for me:
Selection is an active decision.
Hovering is also an active decision but far less so than clicking 'select'. 
So selecting beats hovering.
Nothing selected and nothing active means no actions.  Doing so would feel random.

so:
1. Any image under the mouse cursor is the 'active' image.
2. When one or more images are _selected_, the action works on the selected images. Not the active image.
3. When _no images_ are selected ithe action works on the  _active image_.
4  When there is no image selected and the mouse is somewhere else, there is no selection and not active image. Nothing happens.


The current behaviour is so random and unpredictable that I filed a bug report for it. I just can not work out when rating/color labels are applies to one or all images, I have tried for half an hour and I can not see any logic in its behaviour. 

The art of designing user interfaces is not to cram in as many features as possible and explain it in the documentation;  Really really good userinterfaces do not documentation to explain them.

**Comment by Nilvus** [comment]
I agree with you but I suggest you to post on linked issue and continue discussion there. Maybe one that will change.

**Comment by elstoc** [comment]
Essentially act-on-hover is the thing that needs removing. We could probably cherry-pick some commits from R&dt, which iirc already has that removed.

**Comment by rekcodocker** [comment]
> 



> Essentially act-on-hover is the thing that needs removing. We could probably cherry-pick some commits from R&dt, which iirc already has that removed.

To be clear: 
You mean you could no longer apply any function when you hover over a picture?  
To do away entirely with the notion of an 'active image' ?  Hovering means nothing?

(That I find very scary. That is extremely useful behaviour)

**Comment by elstoc** [comment]
> You mean you could no longer apply any function when you hover over a picture?

Yes, it's just far too easy to accidentally do stuff IMO. I would be fine with a tick box to enable it, but it should default to off as it is too much of a surprise to new users.

**Comment by quovadit** [comment]
> I would be fine with a tick box to enable it, but it should default to off as it is too much of a surprise to new users.

That's what I meant with the two modes (see comment [above](https://github.com/darktable-org/darktable/issues/6025#issuecomment-1264714572))

**Comment by Solarer** [comment]
Most of the stuff has already been done my me and is sitting idle in a branch waiting for clean up and merge

* Fix order of acting so that selection has hingest priority
* Allow to enable/disable mouse over for people who want it
* keep old order of acting as a fall back so that people can switch back if they want (I would prefer to just delete that code)
* all of this can be tuned via a selector in the settings screen

Problem is that this requires a change to the culling view mode and my first draft was not approved by @AlicVB . Since then I have been very busy with life and it will stay that way for the unforseeable future. So I cannot complete the requested changes.

If someone is willing to put in the time to complete this I would be happy to assist and explain the new code...

**Comment by airflow2010** [comment]
I read this discussion with great interest. While I cannot really contribute something very meaningful to it, I just want to raise my hand signalling that I think the implementation of image selection in darktable leaves to be desired, and frustrates me a lot. I recently tried to describe the problems I have with it in this thread: https://discuss.pixls.us/t/frustration-with-culling-mode/33795

What I think is interesting about the discussion here is that I notice that different users have different problems with consistent behaviour of image selection, leading to mistakes and unexpected results.
1) Some people are accidently manipulating the wrong image because they hovered over an image, making the existing selection invalid - they would want the "hover"-selection to be disabled.
2) Some people (like me) are using the "hover"-selection method in the culling mode (where it is mandatory, because there you cannot select pictures by clicking on them compared to lighttable-mode), and are frustrated because sometimes the hover-mechanism doesn't work reliably, and then more images are manipulated then intended - this is described in detail [here](https://github.com/darktable-org/darktable/issues/13014) or in the discussion thread i linked in the first paragraph.

**Comment by elstoc** [comment]
The worst offender, though, is when you purposely select multiple images (with shift+click and/or ctrl+click) and then accidentally apply changes to an unselected image just because you happened not to be hovering your mouse over any of your chosen images at the time you pressed the shortcut.

**Comment by airflow2010** [comment]
> Most of the stuff has already been done by me and is sitting idle in a branch waiting for clean up and merge
> 
> * Fix order of acting so that selection has hingest priority
> * Allow to enable/disable mouse over for people who want it
> * keep old order of acting as a fall back so that people can switch back if they want (I would prefer to just delete that code)
> * all of this can be tuned via a selector in the settings screen
> 
> Problem is that this requires a change to the culling view mode and my first draft was not approved by @AlicVB . Since then I have been very busy with life and it will stay that way for the unforseeable future. So I cannot complete the requested changes.
> 
> If someone is willing to put in the time to complete this I would be happy to assist and explain the new code...

@Solarer, thanks for your work on this one. Here in the discussion, and also your coding efforts. Your code was not accepted - may I ask what were the problems with your PR according to @AlicVB? Perhaps we can discuss and find possible solutions that make everybody happy, perhaps even find somebody who can implement it. :-) What would interest me in your suggested solution is, how is your suggested proposal fixing the problem some people have with culling mode (where "mouse-over"-action is a must and where we have many reports of people accidentally rating all pictures on the screen although they hovered over an image).

**Comment by Solarer** [comment]
> where we have many reports of people accidentally rating all pictures on the screen although they hovered over an image

I mainly use culling in single image mode so I never had that issue. I might be mistaking but I think there is always exactly one image active in culling, no matter what you do with the mouse (if not, it's a bug). The only case in which you could edit multiple images by accident is if you hover over one of your selected images in the filmstrip. That will prioritize the full selection over any other image and is a very unintuitive "feature" which is only required because otherwise selection would always be ignored in favor of mouse over. It can be removed if the code is changed so that selection gets a higher priority than mouse over.

If this is indeed the problem that you mean, my code fixes that by making that "feature" redundant and removing it.

> where "mouse-over"-action is a must

You mean you MUST use the mouse in that mode? We could improve keyboard support by making it so that the arrow keys move the white selection border and we only display new images if the first image is selected and we hit the left key or if we select the last visible image and hit the right key. This is not addressed in my code.

> may I ask what were the problems with your PR

My Images to act on code was not the problem. It was a change to culling (which is a prerequisite so that selection can continue to properly work in culling) that was not accepted. 
I made some button layout changes that were more than the bare minimum and @AlicVB wanted to keep the change as small as possible to not confuse people. We found a compromise but I ran out of time to implement it. And also, ran a bit out of motivation because I think my change was a quite elegant solution that would now need to be redone completely.
The culling change is here: https://github.com/darktable-org/darktable/pull/11112

If someone has time to make culling ready for the change to images to act on, we can merge the new images to act on code with little additional work I think.

**Comment by airflow2010** [comment]
The more I read and think about certain design decisions around darktable, the more it becomes clear that it can be difficult sometimes, as methods of thinking and working vary greatly between persons. This problem becomes worse the more complex the program is. And darktable is quite complex. I think it's important to 1) respect other's peoples usecases and viewpoints and 2) be very cautious about adding more complexity.

> I mainly use culling in single image mode so I never had that issue.

This statement alone surprises me. Because in my understanding, culling mode shows its strength when comparing multiple images side by side. In my understanding, that's what it was built for, and how I use it. If I would just want to inspect a single image and rate it then, I would do it from lighttable-view.

> I might be mistaking but I think there is always exactly one image active in culling, no matter what you do with the mouse (if not, it's a bug).

Yes that would be great, but it's not the case. Perhaps you assume that because you use culling mode with single image only, where that would be correct? Otherwise, me and other people wouldn't complain about problems with accidentally rating a bunch of pictures in culling mode instead of only the one they wanted to act on. Of course, if you use culling mode with multiple displayed images, there has to be a method to define *which* image you want to act on. I use the mouse for that.

> The only case in which you could edit multiple images by accident is if you hover over one of your selected images in the filmstrip. That will prioritize the full selection over any other image and is a very unintuitive "feature" which is only required because otherwise selection would always be ignored in favor of mouse over. It can be removed if the code is changed so that selection gets a higher priority than mouse over.
> 
> If this is indeed the problem that you mean, my code fixes that by making that "feature" redundant and removing it.

That's not the problem I mean. The problem I mean is when the mouse is clearly and definitely over a specific image in culling mode (when multiple images are shown besides each other) and want to act on it. In this case, it often (not always) happens that all displayed images are affected, instead of just the one where the mouse was over.

> My Images to act on code was not the problem. It was a change to culling (which is a prerequisite so that selection can continue to properly work in culling) that was not accepted. I made some button layout changes that were more than the bare minimum and @AlicVB wanted to keep the change as small as possible to not confuse people. We found a compromise but I ran out of time to implement it. And also, ran a bit out of motivation because I think my change was a quite elegant solution that would now need to be redone completely. The culling change is here: #11112
> 
> If someone has time to make culling ready for the change to images to act on, we can merge the new images to act on code with little additional work I think.

Thanks for sharing. I read through the discussion, but I'm afraid I miss some of the knowledge about different usecases of culling mode and cannot contribute. I will look into it more, perhaps I can make a suggestion at some point.

**Comment by Solarer** [comment]
> The problem I mean is when the mouse is clearly and definitely over a specific image in culling mode (when multiple images are shown besides each other) and want to act on it. In this case, it often (not always) happens that all displayed images are affected, instead of just the one where the mouse was over.

I have never observed that. What does happen is: When you enter culling, there is no image active initially until you hover the mouse or use the arrow keys. So in this case you would act on all visible images. Once you moved the mouse the white border appears and then you should only act on that one image. This issue was described earlier (e.g. here: #7254) and I believe I also fixed that somewhere in this PR (not sure though). If that is still not it, you need to click around a little bit until you figure out the exact circumstance that causes this behavior.

**Comment by airflow2010** [comment]
> > The problem I mean is when the mouse is clearly and definitely over a specific image in culling mode (when multiple images are shown besides each other) and want to act on it. In this case, it often (not always) happens that all displayed images are affected, instead of just the one where the mouse was over.
> 
> I have never observed that. What does happen is: When you enter culling, there is no image active initially until you hover the mouse or use the arrow keys. So in this case you would act on all visible images - but only until you move around! Once you moved the mouse the white border appears and then you should only act on that one image.

Your description about how the selection process works in this scenario meets exactly my understanding and expectation how it should work. Sadly, even if the white border is displayed, sometimes all images on the screen are affected. As an engineer myself (different area), I understand how important and helpful reproducibility of a given problem is, to narrow it down and finally fix it. But even as I run into this problem constantly I couldn't yet find a way to reproduce it on purpose every time. My guess at the moment is that it's a timing issue and that it happens if you work a bit faster.

I can just encourage one to try to work for just 5 minutes in the way I described, and I am very confident that you'll run into the problem. Alternatively, I will gladly record a screencapture-movie of my workflow and cut it down to a moment where you'll nicely and clearly observe the problem.

Another approach to the problem - actually all of the problems mentioned in this thread - what do you think: How about making the process of selecting the images about which to act on configurable? The OP made a nice ASCII-drawing at the top of this thread, describing the multiple ways about how the program makes this decision. What if *each* of the lines describing one method ("mouse over", "mouse inside table", etc) could be dis/enabled to fit everyone's needs? People who dislike mouse-over selection could disable it, and me and others could fix their problems as well? And the old argument "we cannot change it, because other people rely on it" would be obsolete, because if you don't mess around with this setting, it's just the old way some are used to.

**Comment by AlicVB** [comment]
> I can just encourage one to try to work for just 5 minutes in the way I described, and I am very confident that you'll run into the problem. Alternatively, I will gladly record a screencapture-movie of my workflow and cut it down to a moment where you'll nicely and clearly observe the problem.

that would be great, because what you describe is clearly a bug. Can you open a new issue for that, thought, as it's not related to the discussion here and should be fixed independently of the act-on algorithm... Thanks !

**Comment by airflow2010** [comment]
> that would be great, because what you describe is clearly a bug. Can you open a new issue for that, thought, as it's not related to the discussion here and should be fixed independently of the act-on algorithm... Thanks !

Sure! Here you have two examples of the same problem. In the [first example](https://youtu.be/KSKJKnIDKG0), I use the zoom-feature. In the [second example](https://youtu.be/rkHYabxZb2I), I don't. But the principle is the same. I used an additional software ("Screenkey") to display the keys I pressed during recording, so you can see exactly what I'm doing. In the comments of the video you see a more detailed explanation with the info of exact moment of problem.

I will open a new issue for that as requested, please just let me briefly know that you still agree that's actually a bug and not some kind of misunderstanding from my side.

**Comment by AlicVB** [comment]
yep, seems like a bug... When you fill the new issue, can you provide a step by step as detailed as possible, so we try to reproduce (I can't actually) Thanks !

**Comment by rekcodocker** [comment]
> The worst offender, though, is when you purposely select multiple images (with shift+click and/or ctrl+click) and then accidentally apply changes to an unselected image just because you happened not to be hovering your mouse over any of your chosen images at the time you pressed the shortcut. That behaviour is just plain dumb

True, that makes no sense ever. 
Best (most logical) behaviour is:
- if you have something selected, then hovering does nothing.
- if you have nothing selected, then hovering will determine where an action takes place.

I.e. clicking is a firm positive confirmation.  Moving your mouse over something is also a meaningful action but should never override clicking.

**Comment by ptilopteri** [comment]
since inception this selection method has been the default in dt and I have become
accustomed.  but as an alternative, why not consider the image below the mouse as
"selected" unless selected images exist?
maybe that can be the best of both worlds?

**Comment by Nilvus** [comment]
Just a thing, another issue related to that one: #14462. Maybe this should change and I'm with @elstoc here to change that. This is really needed to improve UI and user-friendly app. Just be like others apps here. I hope this could change for 4.6. @Solarer: you have worked on that (and culling mode). Possibility to see updates on your work?

**Comment by Solarer** [comment]
@Nilvus I created a PR in January for @AlicVB to review but I got no feedback so far. It is not perfect yet but I wanted to discuss the overall idea before spending more time on it.
The PR is a prerequisite for this discussion and does not solve the act-on problems yet.

You can find it here: #13278
I will rebase the code after I received an 'ok' to proceed or feedback what to change.

**Comment by gi-man** [comment]
I would love the option to disable the Act On mouse hovering. I keep rejecting the wrong image (not the one selected, but the one the mouse is at).

**Comment by rekcodocker** [comment]
>  This is really needed to improve UI and user-friendly app. Just be like others apps here.

Exactly this. Maybe the world is not perfect but at least it's consistent. Sometimes it is more userfriendly to be like other apps.

**Comment by oijn** [comment]
If I forget to "park" my mount pointer off screen somewhere - when I select images in lighttable with arrow keys and spacebar - when the displayed image table is shifted as the page scrolls - the highlight jumps to the image under the mouse pointer and not the logical next image. This just add hundreds or on a large library thousands of extra arrow key presses just to go over the images you've already arrowed over and can add ten plus minutes to a simple selection task.

For the last many months - I just thought that this was a specific darktable bug - well at least I know now that this is a unique darktable feature.

**Comment by elstoc** [comment]
> For the last many months - I just thought that this was a specific darktable bug - well at least I know now that this is a unique darktable feature.

Honestly for this one, I'd be tempted to use the term "intentional bug". I guess I'm used to it now after many years but it's still wrong.

**Comment by Solarer** [comment]
@elstoc and @Nilvus, I am still waiting for a review by @AlicVB on my PR but he does not seem to be very active on github since the beginning of the year. Do you know if he will be back soon or can someone else give feedback to the code?

**Comment by AlicVB** [comment]
I'm sorry to be so unavailable these days : my "real" job and some graduations are taking up all my free time...
Especially in this case, as this review requires lot of time and thinking.

I still hope to do it this month (but I've already said that last month...)

**Comment by londeril** [comment]
Hello all, 

### Introduction
Longtime Lightroom Classic user here - I've only very recently looked into Darktable and I'm currently in the process of switching over to it because I love the image pipeline that much going back to Lightroom to try things out feels "stale" and quite "limiting" - so first of all THANKS to everyone who is or has written code for this amazing project!

### The issue
I've been searching around the interwebs and this github because I found the "Act-on-hover" thing DT does quite disorienting. I understand the logic behind it and I can see the benefits of it (I'm a "focus follows mouse" guy since forever) so I can get my head around the fact that the mouse hover selects an image (though I'd still prefer it to be an option to turn this off and have "normal" click to select functionality). 
BUT - the "Act-on-hover" gets very weird and I think even "workflow impeding" when using the arrow keys to navigate the Lighttable view. The way I like to use the Lighttable is to have rather big thumbnails for a quick overview in my culling process and use the arrow keys to select the next image or go back to the previous one. this works fine as long as the Lighttable view stays "on the same page" as soon as the images get "pushed up" (or down) to make room for the next row - and the mouse hovers over an image (like in the middle of the screen) that image gets selected and pressing the arrow keys will select the next image starting from there... this is very confusing... and now that I write it down I realize that it's also quite hard to show in text... so I made a screen capture off it (with sample images to for privacy concerns, naturally):
https://streamable.com/ze5n3g

This can be mitigated by simply moving the mouse out of the Lighttable (over a module, or over the window decoration) but I feel like this should not be necessary.

Yeah... well... in the end I just wanted to add this to the conversation - forgive me if that has been discusses before - this issue is open for a long time, I see ;)

**Comment by Solarer** [comment]
@londeril thanks for sharing your finding. The issue you describe is not actually caused by the act-on code, so it can be fixed independently :wink: 

Please open a new ticket and add your video there

**Comment by londeril** [comment]
@Solarer will do :)

**Comment by ralfbrown** [comment]
Please post any further discussion in #16850.
